### PR TITLE
Replace all pattern match on boolean value by if/else block.

### PR DESCRIPTION
**Pull request by satendrakumar**
Replaced all pattern match on boolean value by if/else block.

[KAFKA-3768](https://issues.apache.org/jira/browse/KAFKA-3768)

### CODE REVIEW

File: core/src/main/scala/kafka/controller/KafkaController.scala [hunk 1]
```diff
@@ -1419,7 +1424,8 @@ class PreferredReplicaElectionListener(controller: KafkaController) extends IZkD
 
   /**
    * Invoked when some partitions are reassigned by the admin command
-   * @throws Exception On any error.
+    *
+    * @throws Exception On any error.
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/KafkaController.scala:1428]: These scaladoc changes should not be here.

File: core/src/main/scala/kafka/controller/PartitionStateMachine.scala [hunk 2]
```diff
@@ -245,11 +247,11 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
...
           // else, check if the leader for partition is alive. If yes, it is in Online state, else it is in Offline state
-          controllerContext.liveBrokerIds.contains(currentLeaderIsrAndEpoch.leaderAndIsr.leader) match {
-            case true => // leader is alive
-              partitionState.put(topicPartition, OnlinePartition)
-            case false =>
-              partitionState.put(topicPartition, OfflinePartition)
+          if(controllerContext.liveBrokerIds.contains(currentLeaderIsrAndEpoch.leaderAndIsr.leader)) {
+            // leader is alive
+            partitionState.put(topicPartition, OnlinePartition)
+          }else{
```
Comment (ijuma) [inline core/src/main/scala/kafka/controller/PartitionStateMachine.scala:253]: This is not following our formatting guidelines, there should be a space after `if` and a space before and after `else`.

File: core/src/main/scala/kafka/message/ByteBufferBackedInputStream.scala [hunk 3]
```diff
@@ -21,22 +21,23 @@ import java.io.InputStream
...
 class ByteBufferBackedInputStream(buffer:ByteBuffer) extends InputStream {
+
   override def read():Int  = {
-    buffer.hasRemaining match {
-      case true =>
-        (buffer.get() & 0xFF)
-      case false => -1
-    }
+    if(buffer.hasRemaining)
+      (buffer.get() & 0xFF)
```
Comment (ijuma) [inline core/src/main/scala/kafka/message/ByteBufferBackedInputStream.scala:27]: No need for parenthesis here.

File: core/src/main/scala/kafka/tools/JmxTool.scala [hunk 4]
```diff
@@ -118,7 +118,8 @@ object JmxTool extends Logging {
     }
   }
 
-  def queryAttributes(mbsc: MBeanServerConnection, names: Iterable[ObjectName], attributesWhitelist: Option[Array[String]]) = {
+  def queryAttributes(mbsc
+                      : MBeanServerConnection, names: Iterable[ObjectName], attributesWhitelist: Option[Array[String]]) = {
```
Comment (ijuma) [inline core/src/main/scala/kafka/tools/JmxTool.scala:122]: This change seems to be unrelated to the PR.

**Comment by ijuma** [comment]
Thanks for the PR @satendrakumar06. It is certainly better to use `if/else` for such cases.

I left a few comments. Can you please review the whole PR and make sure it doesn't have unrelated changes and that the changes follow Kafka's code conventions? Once you do that, there will be less noise and I can do a better review.

**Comment by satendrakumar** [comment]
@ijuma Thanks for quick review.  Sorry for inconvenience. Definitely, I will do these changes and I should  follow Kafka's code conventions.

**Comment by satendrakumar** [comment]
@ijuma Is any way to verify Kafka's code conventions before creating pull request ?
Like:
`./gradlew checkstyleMain checkstyleTest`
It is only for java.

**Comment by ijuma** [comment]
@satendrakumar06, not yet.

**Comment by satendrakumar** [comment]
@ijuma I  have removed all unnecessary changes(that was my IDE fault). Created new [pull request](https://github.com/apache/kafka/pull/1445)

Should i close this ?

**Comment by ijuma** [comment]
Yes, please close it. In the future, you can update the same PR instead of creating a new one.

**Comment by satendrakumar** [comment]
@ijuma Sorry , I have already created [PR](https://github.com/apache/kafka/pull/1445). It seems  OK because all test showing passed. 

If you want I can update this PR. But I need to close [PR#1445](https://github.com/apache/kafka/pull/1445). 

What you suggest ?
